packages/gasket-helper-intl/lib/index.d.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Locale settings and known locale file paths
+ */
+export interface LocaleManifest {
+    /**
+     * Base URL where locale files are served
+     */
+    basePath?: string
+    /**
+     * Path to endpoint with JSON files
+     */
+    localesPath: LocalePathPart
+    /**
+     * Locale to fallback to when loading files
+     */
+    defaultLocale: Locale
+    /**
+     * Mapping of locales to share files
+     */
+    localesMap?: Record<Locale, Locale>
+    /**
+     * Available locale files to content hashes
+     */
+    paths: Record<LocalePath, string>
+}
+
+/**
+ * Partial URL representing a directory containing locale .json files
+ * or a URL template with a `:locale` path param to a .json file.
+ */
+export type LocalePathPart = string;
+/**
+ * URL path to a locale .json file
+ */
+export type LocalePath = string;
+/**
+ * Language code only
+ */
+export type Lang = string;
+/**
+ * Language code with region
+ */
+export type Locale = Lang;
+/**
+ * State of loaded locale files
+ */
+export interface LocalesState {
+    messages: Record<string, string>
+    status: Record<LocalePath, LocaleStatus>
+}
+
+/**
+ * Enum for local status values
+ */
+export type LocaleStatus = string;
+export namespace LocaleStatus {
+    const LOADING: string;
+    const LOADED: string;
+    const ERROR: string;
+}
+
+/**
+ * Props for a Next.js page containing locale and initial state
+ */
+export interface LocalesProps extends LocalesState {
+    locale: Locale
+}
+
+/**
+ * @classdesc Utility class for loading locale files
+ *
+ * @param {Object} config - Configuration
+ * @param {LocaleManifest} config.manifest - Locale file manifest
+ * @param {string} [config.basePath] - Locale file base path. Defaults to `manifest.basePath`
+ * @constructor
+ */
+export function LocaleUtils(config: {
+    manifest: LocaleManifest;
+    basePath?: string;
+}): void;
+export class LocaleUtils {
```

Comment:
It's odd to have two exports with the same name like this. I'm not sure how TS handles this, but it might be that one replaces the other. I'm guessing you want `LocaleUtils` to work whether called with `new` or not? Perhaps you can just settle on TS users using the `new` option since it shouldn't be too large of an ask for that code to be changed.

Reply:
Yea, I thought this was weird too but it's what the generator gave me. Thanks for the example!